JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -4,40 +4,65 @@ Given 2 strings write a function that will check if it is
 1 or 0 edits away*/
 
 const isOneAway = (str1, str2) => {
+  const diffInLen = Math.abs(str1.length - str2.length);
+  if (diffInLen > 1) {
+    return false;
+  }
+
+  let errorCount = 0;
   if (str1.length === str2.length) {
-    let errorCount = 0;
     for (let i = 0; i < str1.length; i++) {
       if (str1[i] !== str2[i]) {
         errorCount++;
-      }
-      if (errorCount > 1) {
-        return false;
+        if (errorCount > 1) {
+          return false;
+        }
       }
     }
   } else {
-    let errorCount = 0;
-    const longestStr = findLongestStr(str1, str2);
-    let x = 0;
-    for(let i = 0; i < longestStr; i++) {
-      if (str1[i] !== str2[x]) {
+    const longStr = str1.length > str2.length ? str1 : str2;
+    const shortStr = str1.length < str2.length ? str1 : str2;
+    for (
+      let i = 0, x = 0;
+      i < longStr.length && x < shortStr.length;
+      i++, x++
```

Comment:
given that at any point in time `i === x + errorCount`, would it be possible to just use two variables? also, it is customary to use `i`, `j` for indices. you could alternatively use `longIdx` and `shortIdx` if you want to use more descriptive names, but `i`, `j` is okay.

Reply:
Good catch. There was no need for the variable `x` which also helped reduce the amount of code in this for loop. Also, in regards to the indices, sometimes, I'm not a huge fan of `i` and `j` since they look so similar, but yes, I agree. I'm going to try to use `i` and `j` instead since this is more customary.